chain/client/src/client.rs

```diff
@@ -514,13 +536,11 @@ impl Client {
             &*validator_signer,
             next_bp_hash,
             block_merkle_root,
+            timestamp_override,
         );
 
         // Update latest known even before returning block out, to prevent race conditions.
-        self.chain.mut_store().save_latest_known(LatestKnown {
-            height: next_height,
-            seen: to_timestamp(Clock::utc()),
-        })?;
+        self.chain.mut_store().save_latest_known(LatestKnown { height: next_height, seen: at })?;
```

Comment:
:thinking: I don't like that `at` here at `LatestKnown` and the `timestamp` of the block are different, but this is pre-existing behavior. So, no need to change anything in this PR, but I woneder if the following change makes sense as a follow-up:

```diff
diff --git a/chain/client/src/client.rs b/chain/client/src/client.rs
index c6091c459..f5982fa6b 100644
--- a/chain/client/src/client.rs
+++ b/chain/client/src/client.rs
@@ -451,21 +451,10 @@ impl Client {
             prev_next_bp_hash
         };
 
-        #[cfg(feature = "sandbox")]
-        let delta_time = self.sandbox_delta_time();
-
-        let at = to_timestamp(Clock::utc());
+        #[cfg(not(feature = "sandbox"))]
         let timestamp_override = Option::<u64>::None;
-
-        // Sandbox can override the timestamp normally produced from a Block. This new timestamp
-        // is from when a fast_forward request that has been processed.
         #[cfg(feature = "sandbox")]
-        let (at, timestamp_override) = if delta_time > 0 {
-            let at = at + delta_time;
-            (at, Some(at))
-        } else {
-            (at, timestamp_override)
-        };
+        let timestamp_override = to_timetamp(Clock::now()) + self.sandbox_delta_time();
 
         // Get block extra from previous block.
         let mut block_merkle_tree =
@@ -538,9 +527,11 @@ impl Client {
             block_merkle_root,
             timestamp_override,
         );
+        let latest_known =
+            LatestKnown { height: block.header().height(), seen: block.header().timestamp() };
 
         // Update latest known even before returning block out, to prevent race conditions.
-        self.chain.mut_store().save_latest_known(LatestKnown { height: next_height, seen: at })?;
+        self.chain.mut_store().save_latest_known(latest_known)?;
 
         metrics::BLOCK_PRODUCED_TOTAL.inc();
 
```

Reply:
I ended up changing it to what you suggested, since I had to change it already for swapping the return type of `sandbox_delta_time` to `Duration`. And pretty much same logic at the end of the day too, so it should work the same